### PR TITLE
GSW-1001 refactor: decrease gas consumption

### PR DESCRIPTION
**Pull request by r3v4s**
## New
1. some of (realm)pool's calculation logic has been moved (package) pool
2. remove test files from contract folder to prevent deploy

## Update
3. README.md reflects unit tests and tx-tar testing

### CODE REVIEW

File: pool/position_update.gno [hunk 1]
```diff
@@ -67,11 +67,13 @@ func (pool *Pool) updatePosition(
 
 	if liquidityDelta.IsNeg() {
 		if flippedLower {
-			pool.tickClear(tickLower)
+			//pool.tickClear(tickLower)
```
Comment (notJoon) [inline pool/position_update.gno:70]: leftover?

File: pool/position_update.gno [hunk 2]
```diff
@@ -67,11 +67,13 @@ func (pool *Pool) updatePosition(
...
+			delete(pool.ticks, tickLower)
 		}
 
 		if flippedUpper {
-			pool.tickClear(tickUpper)
+			//pool.tickClear(tickUpper)
```
Comment (notJoon) [inline pool/position_update.gno:75]: leftover?

**Review by notJoon — DISMISSED** [review]

File: pool/position_update.gno (hunk 1 above)
Comment (r3v4s) [reply pool/position_update.gno]: bd23db5ab20b004f5a7a5e5526d807067aa307a2

File: pool/position_update.gno (hunk 2 above)
Comment (r3v4s) [reply pool/position_update.gno]: bd23db5ab20b004f5a7a5e5526d807067aa307a2

**Review by notJoon — APPROVED** [review]

**Pull request merged**